src/app/payments/send-lightning.ts

```diff
@@ -0,0 +1,722 @@
+        )
+      }
+
+      const lndService = LndService()
+      if (lndService instanceof Error) return lndService
+
+      const priceRatio = PriceRatio({
+        usd: paymentFlow.usdPaymentAmount,
+        btc: paymentFlow.btcPaymentAmount,
+      })
+      if (priceRatio instanceof Error) return priceRatio
+      const displayCentsPerSat = priceRatio.usdPerSat()
+
+      const converter = NewDisplayCurrencyConverter(displayCentsPerSat)
+
+      const journal = await LockService().extendLock({ logger, lock }, async () => {
+        const metadata = LedgerFacade.LnSendLedgerMetadata({
+          amountDisplayCurrency: converter.fromUsdAmount(paymentFlow.usdPaymentAmount),
+          feeDisplayCurrency: converter.fromUsdAmount(paymentFlow.usdProtocolFee),
+          displayCurrency: DisplayCurrency.Usd,
+
+          paymentFlow,
+          pubkey: outgoingNodePubkey || lndService.defaultPubkey(),
+          paymentHash,
+          feeKnownInAdvance: !!rawRoute,
+        })
+
+        return LedgerFacade.recordSend({
+          description: paymentFlow.descriptionFromInvoice,
+          amountToDebitSender: {
+            btc: {
+              currency: paymentFlow.btcPaymentAmount.currency,
+              amount:
+                paymentFlow.btcPaymentAmount.amount + paymentFlow.btcProtocolFee.amount,
+            },
+            usd: {
+              currency: paymentFlow.usdPaymentAmount.currency,
+              amount:
+                paymentFlow.usdPaymentAmount.amount + paymentFlow.usdProtocolFee.amount,
+            },
+          },
+          senderWalletDescriptor: paymentFlow.senderWalletDescriptor(),
+          metadata,
+        })
+      })
+      if (journal instanceof Error) return journal
+      const { journalId } = journal
+
+      const payResult = rawRoute
+        ? await lndService.payInvoiceViaRoutes({
+            paymentHash,
+            rawRoute,
+            pubkey: outgoingNodePubkey,
+          })
+        : await lndService.payInvoiceViaPaymentDetails({
+            decodedInvoice,
+            btcPaymentAmount: paymentFlow.btcPaymentAmount,
+            maxFeeAmount: paymentFlow.btcProtocolFee,
+          })
+
+      // Fire-and-forget update to 'lnPayments' collection
+      if (!(payResult instanceof LnAlreadyPaidError)) {
+        LnPaymentsRepository().persistNew({
+          paymentHash: decodedInvoice.paymentHash,
+          paymentRequest: decodedInvoice.paymentRequest,
+          sentFromPubkey: outgoingNodePubkey || lndService.defaultPubkey(),
+        })
+
+        if (!(payResult instanceof Error))
+          LedgerFacade.updateMetadataByHash({
+            hash: paymentHash,
+            revealedPreImage: payResult.revealedPreImage,
+          })
+      }
+      if (payResult instanceof LnPaymentPendingError) {
+        paymentFlow.paymentSentAndPending = true
+        paymentFlowRepo.updateLightningPaymentFlow(paymentFlow)
```

Comment:
Can be changed to `updatePendingLightningPaymentFlow` and this old method removed to clean up `update...` interface